### ISSUE TITLE
Docs: outline for polygons

### ISSUE THREAD

**Issue by dv343** [tangrams/tangram#585]
I was reading https://mapzen.com/documentation/tangram/draw/#outline , and it says that `outline` can only be used for lines and points.

However, there is a reference here (https://mapzen.com/documentation/tangram/draw/#stroke) that mention `outline` for polygons.

Could you clarify the support for outlines with polygons?

Thank you!

**Comment by bcamper** [comment]
Ah, the reference to `outline` in the context of polygons is incorrect (we had direct support for that a long time ago but no longer).

The way to draw an outline for a polygon is to create two `draw` blocks:

```
draw:
   # draw polygon fill
   polygons:
      ...
   # draw polygon outline
   lines:
      width: 1px
```

**Comment by dv343** [comment]
Therefore, I understand that blending would be different, right? In your example, polys will be rendered first, and then lines, no?

May I ask why did you remove support for it?

Thank you again! :slightly_smiling_face:

**Comment by bcamper** [comment]
Blending is not usually applied to "opaque" geometry, unless it is set in the style with the `blend` parameter; however, traditional alpha blending won't behave as expected in Tangram with polygons and lines currently because they rely on the depth buffer for sorting, so you get unexpected blend + depth interactions. (This could theoretically be fixed with a full Order Independent Transparency implementation, but this is not being actively considered at this time due to the complexity/overhead.)

For basic cases where you want typical semi-transparent buildings, we are adding a new blend mode that does a two-pass rendering (it still is not your traditional blending with depth, because only the final front-facing polygons are drawn, but this is similar to the 3d building effects in Google and Apple maps). This is in progress for Tangram ES [here](https://github.com/tangrams/tangram-es/pull/1615) and will be added to JS subsequently as well.

Traditional blending is applied to points and text labels, using the `overlay` blend mode. These don't have any depth, and are meant to be overlaid on top of the 3d scene. They do have a `priority` which is used to resolve collisions, but they do not have any concept of draw order or depth.

As to the draw order of the above, for the opaque geometry using the depth buffer, the `order` parameter will determine which one is on top (it works by making small adjustments to the geometry depth, similar to `glPolygonOffset`).

The polygons-specific outline syntax was removed (a long time ago, before we even had any official releases :) because it was more cumbersome to support the additional "sub-block" parsing (the outline essentially became a second, virtual `draw` group). It has been kept in the case of `lines` because they are more tightly and conceptually coupled (since the outline is actually drawn as a second copy of the geometry underneath, we found it was more confusing and error-prone to require the user to create separate `draw` groups).
